clients/mongoStoreClient_test.go

```diff
@@ -171,10 +189,57 @@ func TestIndexes(t *testing.T) {
 			t.Fatalf("TestIndexes expected [%s] got [%s] ", uploadid_query_idx, idxs[1].Name)
 		}
 	}
+}
+
+func TestGetTimeLastEntryUser(t *testing.T) {
+
+	mc := initTestData()
+
+	entry, err := mc.GetTimeLastEntryUser(valid_groupid)
+
+	t.Logf("TestGetTimeLastEntryUser %s ", entry)
+
+	if len(entry) <= 0 {
+		t.Fatal("GetTimeLastEntryUserAndDevice time entry hsould be set")
+	}
+
+	expectedTime := []byte("2015-01-13T08:44:04.000Z")
+
+	if bytes.Equal(expectedTime, entry) && reflect.DeepEqual(expectedTime, entry) {
+		t.Fatalf("GetTimeLastEntryUser expected [%s] got [%s] ", expectedTime, entry)
+	}
+
+	if err != nil {
+		t.Fatalf("GetTimeLastEntryUser unexpected error [%s]", err.Error())
+	}
+
+}
+
+func TestGetTimeLastEntryUserAndDevice(t *testing.T) {
+
+	mc := initTestData()
+
+	entry, err := mc.GetTimeLastEntryUserAndDevice(valid_groupid, valid_deviceid)
+
+	t.Logf("TestGetTimeLastEntryUserAndDevice %s ", entry)
+
+	if len(entry) <= 0 {
+		t.Fatal("GetTimeLastEntryUserAndDevice time entry hsould be set")
+	}
+
+	expectedTime := []byte("2014-10-28T10:00:00.000Z")
+
+	if bytes.Equal(expectedTime, entry) && reflect.DeepEqual(expectedTime, entry) {
+		t.Fatalf("GetTimeLastEntryUserAndDevice expected [%s] got [%s] ", expectedTime, entry)
+	}
+
+	if err != nil {
+		t.Fatalf("GetTimeLastEntryUserAndDevice unexpected error [%s]", err.Error())
+	}
 
 }
 
-func TestWhereQueryConstruction(t *testing.T) {
+func Test_constructQuery_WhereQueryConstruction(t *testing.T) {
```

Comment:
intentional - bad probably still bad. Its testing the internal method constructQuery with a where. I will have another look and rename as it still doesn't read as I want